platform/parser/xml/src/main/java/org/codice/ddf/parser/xml/XmlParser.java

```diff
@@ -171,58 +174,79 @@ public void marshal(ParserConfigurator configurator, Object obj, Node node)
 
   private void marshal(ParserConfigurator configurator, Consumer<Marshaller> marshallerConsumer)
       throws ParserException {
-    JAXBContext jaxbContext =
-        getContext(configurator.getContextPath(), configurator.getClassLoader());
-
-    ClassLoader tccl = Thread.currentThread().getContextClassLoader();
     try {
-      Thread.currentThread().setContextClassLoader(configurator.getClassLoader());
-      Marshaller marshaller = jaxbContext.createMarshaller();
-      if (configurator.getAdapter() != null) {
-        marshaller.setAdapter(configurator.getAdapter());
-      }
-      if (configurator.getHandler() != null) {
-        marshaller.setEventHandler(configurator.getHandler());
-      }
-      for (Map.Entry<String, Object> propRow : configurator.getProperties().entrySet()) {
-        marshaller.setProperty(propRow.getKey(), propRow.getValue());
-      }
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Void>)
+              () -> {
+                JAXBContext jaxbContext =
+                    getContext(configurator.getContextPath(), configurator.getClassLoader());
+
+                ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+                try {
+                  Thread.currentThread().setContextClassLoader(configurator.getClassLoader());
+                  Marshaller marshaller = jaxbContext.createMarshaller();
+                  if (configurator.getAdapter() != null) {
+                    marshaller.setAdapter(configurator.getAdapter());
+                  }
+                  if (configurator.getHandler() != null) {
+                    marshaller.setEventHandler(configurator.getHandler());
+                  }
+                  for (Map.Entry<String, Object> propRow :
+                      configurator.getProperties().entrySet()) {
+                    marshaller.setProperty(propRow.getKey(), propRow.getValue());
+                  }
 
-      marshallerConsumer.accept(marshaller);
-    } catch (RuntimeException | JAXBException e) {
+                  marshallerConsumer.accept(marshaller);
+                } finally {
+                  Thread.currentThread().setContextClassLoader(tccl);
+                }
+                return null;
+              });
+    } catch (final RuntimeException e) {
       LOGGER.debug(MARSHALLING_ERROR_MSG, e);
       throw new ParserException(MARSHALLING_ERROR_MSG, e);
-    } finally {
-      Thread.currentThread().setContextClassLoader(tccl);
+    } catch (final PrivilegedActionException e) {
+      LOGGER.debug(MARSHALLING_ERROR_MSG, e);
+      throw new ParserException(MARSHALLING_ERROR_MSG, e.getException());
     }
```

Comment:
Any `RuntimeException`s will bubble out of the `doPrivileged`, but checked exceptions are wrapped in a `PrivilegedActionException`.